simulator/replayer/replayer.go

```diff
@@ -0,0 +1,101 @@
+package replayer
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path"
+
+	"golang.org/x/xerrors"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/recorder"
+)
+
+type Service struct {
+	applier   ResourceApplier
+	recordDir string
+}
+
+type ResourceApplier interface {
+	Create(ctx context.Context, resource *unstructured.Unstructured) error
+	Update(ctx context.Context, resource *unstructured.Unstructured) error
+	Delete(ctx context.Context, resource *unstructured.Unstructured) error
+}
+
+type Options struct {
+	RecordDir string
+}
+
+func New(applier ResourceApplier, options Options) *Service {
+	return &Service{applier: applier, recordDir: options.RecordDir}
+}
+
+func (s *Service) Replay(ctx context.Context) error {
+	files, err := os.ReadDir(s.recordDir)
+	if err != nil {
+		return xerrors.Errorf("failed to read record directory: %w", err)
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		filePath := path.Join(s.recordDir, file.Name())
+		if err := s.replayRecordsFromFile(ctx, filePath); err != nil {
+			return xerrors.Errorf("failed to replay records from file: %w", err)
+		}
```

Comment:
How about parsing all the json first? It is quite painful to have not-json files in directories and fail after a long replay.

Reply:
Makes sense. I made that change!